### PR TITLE
Introduce namespace into Airbyte Record Message.

### PR DESCRIPTION
**Pull request by davinchia**
## What
Make namespace a field class concept in Airbyte Protocol. This allows the source to propagate namespace and destinations to write to a source-defined namespace. Also sets us up for future namespace related configurability.

## How
* Add an optional `namespace` field to the `AirbyteRecordMessage`. This field will be set by sources that support namespace.
* Introduce `AirbyteStreamNameNamespacePair` as a type-safe manner of identifying streams throughout our code base.
* Convert all internal stream comparison and indexing to use `AirbyteStreamNameNamespacePair` instead of a string comparison. Propagate this to all JDBC source and destinations.
* Convert all tests to properly support namespace.

## Pre-merge Checklist
- [x] *Run integration test for sources and destinations*

## Recommended reading order
1. `airbyte_protocol.yaml` (understand the new field)
2. `BufferedStreamConsumer` and `JdbcBufferedConsumerFactory` (understand destination side changes)
3. `AbstractJdbcSource` (understand source side changes)
4. `JdbcStateManager` and `StateDecoratingIterator` (understand incremental changes)
2. `TestDestination` (main test file to read to understand test changes)

### CODE REVIEW

**Comment by davinchia** [comment]
There is one test case (multiple schema, same table name) that is still failing. This will involve do the same (changing how we index streams in the python code).

Will follow up with another PR into the same branch.

File: airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination/buffered_stream_consumer/BufferedStreamConsumer.java [hunk 1]
```diff
@@ -139,12 +144,14 @@ protected void acceptTracked(AirbyteRecordMessage message) {
 
     // ignore other message types.
     final String streamName = message.getStream();
-    if (!streamNames.contains(streamName)) {
+    final AirbyteStreamNameNamespacePair pair = AirbyteStreamNameNamespacePair.fromRecordMessage(message);
+    LOGGER.info("pair: {}", pair);
```
Comment (cgardens) [inline airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination/buffered_stream_consumer/BufferedStreamConsumer.java:148]: do we still know need this log line? it's going to log alot. once per record message.

**Review by cgardens — APPROVED** [review]

File: airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination/buffered_stream_consumer/BufferedStreamConsumer.java (hunk 1 above)
Comment (davinchia) [reply airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination/buffered_stream_consumer/BufferedStreamConsumer.java:148]: nope. good catch. removing!

**Pull request merged**